### PR TITLE
hud: if hud not running, don't generate the whole view

### PR DESCRIPTION
**Pull request by maiamcc**
this fixes a small amt of the perf problems in #2597, tho not the bulk of them.
unsure if it's worth the complexity, figured I'd toss up a PR. thoughts?

### CODE REVIEW

**Comment by nicks** [comment]
can you give a bit more detail on what this addresses?

The `isRunning` field is _supposed_ to be for handling special cases where the HUD is in the middle of being set up or torn down. A frequent mistake we've made in the past is "oops! we hit a fatal error, then screwed up the teardown codepath, so that the user can't even see the error".

If `isRunning` is false when Tilt is in steady-state, that would surprise me, and is something we should dig into more.

**Comment by nicks** [comment]
(admittedly, `isRunning` is not a great name for this field)

**Comment by maiamcc** [comment]
Oh sorry, this is for cases when `--hud=false` -- `hud.OnChange` is eating up a non-trivial amount of CPU even when the hud is disabled via flag. This attempts to fix that by not doing the whole view conversion if the hud is disabled. That said, idk if this is the right way to go about it or if the CPU savings are worth the complexity introduced. thoughts?

**Comment by nicks** [comment]
can we make it so that the HUD doesn't get installed at all if it's not running? or just make OnChange a no-op? i don't totally understand why the HUD needs to do any work at all in this case.

**Comment by maiamcc** [comment]
It's because we smooshed the responsibility for "if we're not doing any rty stuff, print all the logs since the last change to the terminal" into the hud, which isn't great.

```
logLen := log.Len()
if !h.isRunning && h.currentViewState.ProcessedLogByteCount < logLen {
	fmt.Print(log.String()[h.currentViewState.ProcessedLogByteCount:])
}

h.currentViewState.ProcessedLogByteCount = logLen
```

Idk if this properly ought to be a separate subscriber, or what? I can certainly split it out into its own subscriber (or its own implementation of the `Hud` interface??)

**Comment by maiamcc** [comment]
Looking through, it seemed easier to just do the rearch. now -- wdty?

File: internal/hud/hud.go [hunk 1]
```diff
@@ -293,6 +306,22 @@ func (h *Hud) setView(ctx context.Context, view view.View) error {
 	return h.refresh(ctx)
 }
 
+// setViewLogOnly prints the new logs to stdout. (Should only be invoked if !h.isRunning)
+func (h *Hud) setViewLogOnly(ctx context.Context, log model.Log) error {
```
Comment (nicks) [inline internal/hud/hud.go]: i think this function is dead now?

Comment (maiamcc) [reply internal/hud/hud.go]: good catch!

**Comment by maiamcc** [comment]
hudfix before and after

Well this isn't the _most_ useful, but it did something, at least...

**Pull request merged**